graphql/batch_executor.go

```diff
@@ -106,6 +108,24 @@ func resolveWorkUnit(unit *WorkUnit) []*WorkUnit {
 	return units
 }
 
+func resolveBatchUnit(unit *WorkUnit) []*WorkUnit {
+	results, err := safeResolveBatch(unit.ctx, unit.field, unit.sources, unit.selection.Args, unit.selection.SelectionSet)
+	if err != nil {
+		for _, dest := range unit.destinations {
+			dest.Fail(err)
+		}
+		return nil
+	}
+	unitChildren, err := resolveBatch(unit.ctx, results, unit.field.Type, unit.selection.SelectionSet, unit.destinations)
```

Comment:
nit: hmmm. seems we've now overloaded resolve. maybe we have executeWorkUnit (instead of resolveBatchUnit), safeExecute[Batch]Resolver, and then resolveBatch, resolveBatchList? although resolveBatch isn't quite what resolve means either in the graphql spec.